engine/collection/epochmgr/engine.go

```diff
@@ -189,13 +189,16 @@ func (e *Engine) Ready() <-chan struct{} {
 	}, func() {
 		// check the current phase on startup, in case we are in setup phase
 		// and haven't yet voted for the next root QC
-		phase, err := e.state.Final().Phase()
+		finalSnapshot := e.state.Final()
+		phase, err := finalSnapshot.Phase()
 		if err != nil {
 			e.log.Error().Err(err).Msg("could not check phase")
```

Comment:
I think this is an unexpected error,  should we fatal instead?
```suggestion
			e.log.Fatal().Err(err).Msg("could not check phase")
```